### PR TITLE
fix(docx): suppress automatic spacing at cell edges

### PR DESCRIPTION
**Pull request by eliahilse**
TL;DR:

Repro file:
https://docxcorp.us/documents/003732bcbcca8045cbd44c4c11736826005f780a52b5a228845333fef026fa6e.docx

Summary:
- Suppress automatic paragraph spacing at the first and last edges of table cells.
- Preserve explicit spacing and interior gaps, including nested tables, content controls, and subsequent paragraph edits.

Test plan:
- [x] Pass all 23 editing fidelity tests, including spacing provenance, nested content, edits, pagination, and six mixed drawing/text boundary cases.
- [x] Verify the behavior with Word exports of isolated spacing cases. Six square/top-bottom/inline leading/trailing drawing cases have byte-identical automatic/zero-spacing PNGs; explicit spacing changes the control output.
- [x] Check the unchanged source with the same native font inputs: 3 to 1 pages, matching Word.
- [x] Complete an independent review and pass formatting and diff checks.
- [x] Compare the rebuilt browser against the unchanged local Word reference using Oxi RGB scoring: 3 → 1 pages (Word: 1), page-penalized SSIM 0.2190585 → 0.7538852; common-page SSIM 0.6571755 → 0.7538852.
- [x] Pass the [14-file DOCX fidelity measurement](https://github.com/openooxml/betteroffice/actions/runs/34869217886): 14/14 scored, mean 0.8309269; every score and page count unchanged.

Measurements use the local reconstructed Oxi trial and the same Word for Mac reference, source bytes, and pinned CDN fonts. They do not establish a ranking against the published Windows leaderboard.

### CODE REVIEW

**Comment by eliahilse** [comment]
Native and browser validation is complete. The 14-file corpus measurement scored all 14 files, with every score and page count unchanged (mean 0.8309269). Targeted Oxi SSIM/page-count results are in the PR body; the larger 100-file local trial is finishing separately. Current main and the merged dependency update are synchronized, preserving the tested file tree.

@greptileai review

File: crates/docx-edit/src/bridge/mod.rs [hunk 1]
```diff
@@ -2883,6 +2914,40 @@ fn lower_paragraph_border(
...
+    blocks: &mut [LayoutBlock],
+    values: &BTreeMap<String, Any>,
+    edges: CellEdges,
+) {
+    if edges.before
+        && paragraph_auto_spacing(values, "beforeAutospacing")
+        && let Some(spacing) = blocks.iter_mut().find_map(|block| match block {
+            LayoutBlock::Paragraph(paragraph) => paragraph.attrs.as_mut()?.spacing.as_mut(),
+            _ => None,
+        })
```
[the remark this answers is not in the document]
Comment (eliahilse) [reply crates/docx-edit/src/bridge/mod.rs:2936]: Verified this against Word 16.112.4 with six cases: square-wrapped, top-and-bottom-wrapped, and inline shapes, each before and after text in a single cell paragraph. Automatic before/after spacing and explicit zero spacing produce byte-identical 150-DPI PNGs (SHA-256 de763a00873b303e0816fe063b96632bc4c4fe0356804179a83b32262b83fa31). An explicit 12pt-spacing control produces a different image.

The source drawing and text belong to one Word paragraph; splitting them into layout blocks does not introduce an authored paragraph-spacing boundary. Keeping the automatic gap between those synthetic fragments would diverge from Word. Added the six-case regression in 8e29d519; all 23 editing fidelity tests pass. Retaining the current suppression and resolving this as a false positive.

**Comment by eliahilse** [comment]
The drawing/text concern is checked against six Word cases and covered by a new regression. Word produces identical automatic/zero-spacing images, and all 23 editing fidelity tests pass. The renderer is unchanged from the validated 14-file run.

@greptileai review
